flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManagerDriver.java

```diff
@@ -170,8 +159,12 @@ protected void initializeInternal() throws Exception {
 			resourceManagerClient.start();
 
 			final RegisterApplicationMasterResponse registerApplicationMasterResponse = registerApplicationMaster();
-			getContainersFromPreviousAttempts(registerApplicationMasterResponse);
-			updateMatchingStrategy(registerApplicationMasterResponse);
+			int maxPriority = getContainersFromPreviousAttempts(registerApplicationMasterResponse);
+			taskExecutorProcessSpecContainerResourcePriorityAdapter =
+				new TaskExecutorProcessSpecContainerResourcePriorityAdapter(
+					maxPriority + 1,
```

Comment:
Will this be a problem that we continue increasing the priority value? Potentially, this can grow arbitrarily large.

Reply:
Should not be a problem. Yarn does not have constraints on the max priority value, as long as it does not grow lager than `Integer#MAX_VALUE`. In addition, we had this increasing priorities in Blink for years without any problem.